src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -273,10 +273,14 @@ private void checkIfGroupsExists(Course course) {
 
         User user = userService.getUserWithGroupsAndAuthorities();
         List<Exercise> exercises = exerciseService.findAllForCourse(course, false, principal, user);
+
+        exercises = exercises.stream()
+            .filter(exercise -> exercise instanceof TextExercise || exercise instanceof ModelingExercise)
+            .collect(Collectors.toList());
+
```

Comment:
to improve performance: can you filter in the database call, i.e. in the repository using HQL?

Reply:
I don't think so, I don't find any column to use as a way to differentiate among different type of exercises :thinking: What do you suggest?

Reply:
you could use a native query, but I don't think this would improve the performance significantly, so you can keep the code as is

@Query(value = "select * from exercise e WHERE (e.discriminator = 'T' or e.discriminator = 'M') and e.course_id = :#{#courseId}", nativeQuery = true)

Reply:
Then I'd merge like this, instead of creating a super specialized method in the `ExerciseRepository`, because I think `findAllTextAndModellingExercisesByCourseId` is a bit too specific, what do you think?